src/wallet/api/walletApi.ts

```diff
@@ -16,46 +28,68 @@ import Input from '../../models/input';
  */
 
 const walletApi = {
-  getWalletStatus(id: string): Promise<any> {
-    const data = { params: { id } }
-    return axiosInstance().get('wallet', data);
+  async getWalletStatus(wallet: HathorWalletServiceWallet): Promise<WalletStatusResponse> {
+    const axios = await axiosInstance(wallet);
+    return axios.get('wallet/status');
   },
 
-  createWallet(xpubkey: string): Promise<any> {
+  async createWallet(xpubkey: string): Promise<WalletStatusResponse> {
     const data = { xpubkey };
-    return axiosInstance().post('wallet', data);
+    const axios = await axiosInstance();
+    return axios.post('wallet/init', data);
   },
 
-  getAddresses(id: string): Promise<any> {
-    const data = { params: { id } }
-    return axiosInstance().get('addresses', data);
+  async getAddresses(wallet: HathorWalletServiceWallet): Promise<AddressesResponse> {
+    const axios = await axiosInstance(wallet);
+    return axios.get('wallet/addresses');
   },
 
-  getBalances(id: string, token: string | null = null): Promise<any> {
-    const data = { params: { id } }
+  async getNewAddresses(wallet: HathorWalletServiceWallet): Promise<NewAddressesResponse> {
+    const axios = await axiosInstance(wallet);
+    return axios.get('wallet/addresses/new');
+  },
+
+  async getBalances(wallet: HathorWalletServiceWallet, token: string | null = null): Promise<BalanceResponse> {
+    const data = { params: {} };
     if (token) {
       data['params']['token_id'] = token;
     }
-    return axiosInstance().get('balances', data);
+    const axios = await axiosInstance(wallet);
+    return axios.get('wallet/balances', data);
   },
 
-  getHistory(id: string, token: string | null = null): Promise<any> {
+  async getHistory(wallet: HathorWalletServiceWallet, token: string | null = null): Promise<HistoryResponse> {
     // TODO add pagination parameters
```

Comment:
Why is this a TODO? We are not going to paginate it yet?

Reply:
Yes, we are. This PR was supposed to handle only the token actions integration. I had to also do the authorization methods, otherwise it would not work. So I decided not to add any more features and this TODO is already removed and implemented in the PR that matches the faces signatures